src/audio-worklet/basic/audio-worklet-node-options/main.js

```diff
@@ -3,30 +3,44 @@
 // found in the LICENSE file.
 
 const audioContext = new AudioContext();
+let oscillatorProcessor;
 
 const startAudio = async (context, options) => {
   await context.audioWorklet.addModule('oscillator-processor.js');
-  const oscillatorProcessor =
-      new AudioWorkletNode(context, 'oscillator-processor', {
-        processorOptions: {
-          waveformType: options.waveformType,
-          frequency: options.frequency,
-        }});
+  oscillatorProcessor =
+    new AudioWorkletNode(context, 'oscillator-processor', {
+      processorOptions: {
+        waveformType: options.waveformType,
+        frequency: options.frequency,
+      }
+    });
   oscillatorProcessor.connect(context.destination);
 };
 
-// A simplem onLoad handler. It also handles user gesture to unlock the audio
+const stopAudio = () => {
+  if (oscillatorProcessor) {
+    oscillatorProcessor.disconnect();
+    oscillatorProcessor = null;
+  }
+};
+
+// A simple onLoad handler. It also handles user gesture to unlock the audio
 // playback.
 window.addEventListener('load', async () => {
   const buttonEl = document.getElementById('button-start');
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    const waveformType =
+    if (!oscillatorProcessor) { // If audio is not playing, start audio
+      const waveformType =
         document.querySelector('#demo-select-waveform-type').value;
-    const frequency = document.querySelector('#demo-input-frequency').value;
-    await startAudio(audioContext, {waveformType, frequency});
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
+      const frequency = document.querySelector('#demo-input-frequency').value;
+      await startAudio(audioContext, { waveformType, frequency });
```

Comment:
So we're calling `startAudio` for every button click and it means that we're calling `addModule` every time.

If we have to introduce `stop` function, we'll have to handle the context creation and the worklet module loading properly.

The simplest solution would be just suspending and resuming the context and keeping the rest unchanged.

Reply:
So we have updated the code to include a check to ensure that addModule is only called if oscillatorProcessor is not already initialized, preventing multiple calls to addModule. Additionally, we have removed the stopAudio() function and instead disconnected the oscillatorProcessor and set it to null. Since there are no active audio nodes connected to the audio context, the context will naturally suspend. Therefore, we have removed the explicit call to suspend the context.

Reply:
Thanks for updating the code, but I think it's still not correct. After setting `oscillatorProcessor` to `null`, `addModule()` can be called again.

`addModule()` with the same arguments should only be called once throughout the document/frame lifetime. It's similar to JS import; you can't call import with the same file multiple times.

Reply:
Done with the changes. Also, I have make sure to remove the `oscillatorProcessor `to `null` and instead used the `audioContext.suspend()`.

Reply:
Looks good!